runtime/src/accounts_partition.rs

```diff
@@ -0,0 +1,718 @@
+//! Partitioning of the accounts into chunks for rent collection
+use {
+    itertools::Itertools,
+    log::trace,
+    solana_sdk::{
+        clock::{Slot, SlotCount, SlotIndex},
+        pubkey::Pubkey,
+        stake_history::Epoch,
+        sysvar::epoch_schedule::EpochSchedule,
+    },
+    std::{collections::HashSet, mem, ops::RangeInclusive},
+};
+
+// Eager rent collection repeats in cyclic manner.
+// Each cycle is composed of <partition_count> number of tiny pubkey subranges
+// to scan, which is always multiple of the number of slots in epoch.
+pub(crate) type PartitionIndex = u64;
+type PartitionsPerCycle = u64;
+pub(crate) type Partition = (PartitionIndex, PartitionIndex, PartitionsPerCycle);
+pub(crate) type RentCollectionCycleParams = (
+    Epoch,
+    SlotCount,
+    bool,
+    Epoch,
+    EpochCount,
+    PartitionsPerCycle,
+);
+type EpochCount = u64;
+
+fn partition_index_from_slot_index(
+    slot_index_in_epoch: SlotIndex,
+    (
+        epoch,
+        slot_count_per_epoch,
+        _,
+        base_epoch,
+        epoch_count_per_cycle,
+        _,
+    ): RentCollectionCycleParams,
```

Comment:
totally unrelated to the PR...I knew you could destructure struct args in-line. Had never seen it done for tuples, neat!